### ISSUE TITLE
Infinite recursion in development mode

### ISSUE THREAD

**Issue by jtojnar** [parcel-bundler/parcel#7528]
# 🐛 bug report

After updating to Parcel 2.1.0, my app crashes in developer mode with an infinite recursion:

> Uncaught InternalError: too much recursion

The app seems to work just fine in production mode.

## 🎛 Configuration (.babelrc, package.json, cli command)

- This is the update commit: https://github.com/fossar/selfoss/commit/590696761de138f8cf8e061c1fccd67b0f6faec0 (mostly just Parcel and Babel updates).
- `package.json` (using `dev` and `build` scripts for development and production builds respectivelly): https://github.com/fossar/selfoss/blob/590696761de138f8cf8e061c1fccd67b0f6faec0/assets/package.json#L58-L59
- `.parcelrc` https://github.com/fossar/selfoss/blob/590696761de138f8cf8e061c1fccd67b0f6faec0/assets/.parcelrc
- no explicit babel config

## 🤔 Expected Behavior

The app should work in developer mode.

## 😯 Current Behavior

It crashes with infinite recursion:

In Chromium I get the following stack trace
1. ![stack trace 1](https://user-images.githubusercontent.com/705123/148294508-098fa7b6-8382-47e3-a87f-d582da21a50f.png)
2. ![stack trace 2](https://user-images.githubusercontent.com/705123/148294397-1009551e-9987-4933-8e73-6b02d3f7f970.png)
3. 2 is repeated forever

In Firefox:

```
Uncaught InternalError: too much recursion
    _typeof _type_of.js:2
    _typeof _type_of.js:2
   …
```

The source map points me to https://github.com/swc-project/swc/blob/d901b6222f8a77beed64968fcb4764bbacf8c755/packages/swc-helpers/src/_type_of.js#L2 (but that package has not been updated).

## 🔦 Context

Just trying to develop after a routine dependency bump.

## 💻 Code Sample

[`assets/` directory in selfoss repo](https://github.com/fossar/selfoss/tree/590696761de138f8cf8e061c1fccd67b0f6faec0/assets)

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| Parcel           | 2.1.0
| Node             |16.13.1
| npm         |8.1.2
| Operating System |NixOS (Linux)

I deleted `.parcel-cache` and `node_modules` before each test (reverted the update commit few times back and forth).

**Comment by devongovett** [comment]
In 2.1.0, we started transpiling node_modules. This seems to have lead to SWC transpiling its own helpers.

**Comment by npulidom** [comment]
Same here, in my case I use `pnpm` as package manager and [overriding](https://pnpm.io/package_json#pnpmoverrides) all @parcel packages deps to `~2.0.0` it's a [quick fix](https://pastebin.pl/view/9842b5e1) (worked even using parcel 2.1.0 cli); yarn can [resolve](https://classic.yarnpkg.com/lang/en/docs/selective-version-resolutions/) packages the same way.

**Comment by devongovett** [comment]
Should be fixed in v2.1.1.
